### PR TITLE
Value handlers set values based on their scopes (Sylius channel) 

### PR DESCRIPTION
**Pull request by LucaGallinari**
It could be simpler to review the PR by viewing the commit and not the entire diff

Will solve #112

### CODE REVIEW

File: spec/ValueHandler/ImageValueHandlerSpec.php [hunk 1]
```diff
@@ -258,4 +250,107 @@ function it_does_not_remove_image_of_other_variant_on_sylius_if_empty_on_akeneo(
...
+                    '_links' => ['download' => ['href' => 'download-url']],
+                ],
+            ]
+        );
+
+        $apiClient->downloadFile('path/to/a/file.jpg')->shouldHaveBeenCalled();
+        $productVariant->addImage($productImage)->shouldNotHaveBeenCalled();
+        $product->addImage($productImage)->shouldHaveBeenCalled();
+        $productImage->setFile($imageFile)->shouldHaveBeenCalled();
+    }
```
[the remark this answers is not in the document]
Comment (LucaGallinari) [reply spec/ValueHandler/ImageValueHandlerSpec.php:355]: Yes, you are right, we should throw in these cases

File: src/ValueHandler/GenericPropertyValueHandler.php [hunk 2]
```diff
@@ -54,28 +58,43 @@ public function handle($subject, string $attribute, array $value): void
...
+
+        $productChannelCodes = array_map(static function (ChannelInterface $channel): ?string {
+            return $channel->getCode();
+        }, $product->getChannels()->toArray());
+        $productChannelCodes = array_filter($productChannelCodes);
+
+        foreach ($value as $valueData) {
+            if (array_key_exists('scope', $valueData) && $valueData['scope'] !== null && !in_array($valueData['scope'], $productChannelCodes, true)) {
+                continue;
+            }
```
[the remark this answers is not in the document]
Comment (LucaGallinari) [reply src/ValueHandler/GenericPropertyValueHandler.php:76]:
```suggestion
            if (array_key_exists('scope', $valueData)) {
                // throw ..
            }
            if ($valueData['scope'] !== null && !in_array($valueData['scope'], $productChannelCodes, true)) {
                continue;
            }
```

**Review by lruozzi9 — COMMENTED** [review]
I think you could upgrade Node version to 12 in GH action... We don't add any JS files in our plugin

File: src/ValueHandler/GenericPropertyValueHandler.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (mmenozzi) [reply src/ValueHandler/GenericPropertyValueHandler.php:76]: Mmm no, in this way we throw if there's the scope key which is one of the key returned by Akeneo. Am I right?

[the remark this answers is not in the document]
Comment (LucaGallinari) [reply src/ValueHandler/GenericPropertyValueHandler.php:76]: 🤦🏻‍♂️  ... fixed

**Review by mmenozzi — APPROVED** [review]

**Pull request merged**